azurerm/internal/services/web/data_source_function_app_host_keys.go

```diff
@@ -0,0 +1,88 @@
+package web
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmFunctionAppHostKeys() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmFunctionAppHostKeysRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"master_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"function_keys": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Sensitive: true,
+			},
```

Comment:
rather than exposing this as a Map - can we lookup the data in a defensive fashion and output those specific fields, so that users don't have to deal with breaking changes in the response format here e.g.

```
defaultFunctionKey := ""
if v, ok := keys["default"]; ok {
  defaultFunctionKey = v
}
d.Set("default_function_key", defaultFunctionKey)
```